processing/src/main/java/org/apache/carbondata/processing/loading/sort/SortStepRowHandler.java

```diff
@@ -192,6 +196,16 @@ public SortStepRowHandler(SortParameters sortParameters) {
     return out;
   }
 
+  public Object[] convertToFlatRow(IntermediateSortTempRow sortTempRow) {
+    List<Object> out = new ArrayList<>();
```

Comment:
Suggest to create with initialCapacity as the number of elements are at this point.

Reply:
@VenuReddy2103 : This is WIP PR, I am working on it now. As effort is more.